### PR TITLE
feat: Update for latest SQL timezone behavior

### PR DESCRIPTION
**Pull request by jeffreyssmith2nd**
Closes #

_Describe your proposed changes here._

- [ ] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
  ([if necessary](https://github.com/influxdata/docs-v2/blob/master/CONTRIBUTING.md#sign-the-influxdata-cla))
- [ ] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md [hunk 1]
```diff
@@ -1072,3 +1073,70 @@ SELECT
...
+  '2024-04-01T00:00:20Z'::timestamp AT TIME ZONE 'Europe/Brussels' as time_timestamp,
+  tz('2024-04-01T00:00:20', 'Europe/Brussels') as time_tz;
+```
+
+| time_timestamp               | time_tz                    |
+| :--------------------------- | :------------------------- |
+| 2024-04-01T00:00:20+02:00    | 2024-04-01T02:00:20+02:00  |
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md:1142]:
~~~suggestion
##### Differences between tz and AT TIME ZONE

`tz` and [`AT TIME ZONE`](/influxdb/cloud-dedicated/reference/sql/operators/other/#at-time-zone)
differ when the input timestamp **does not** have a timezone.

- When using an input timestamp that does not have a timezone (the default behavior in InfluxDB) with the
  `AT TIME ZONE` operator, the operator returns the the same timestamp, but with a timezone offset
  (also known as the "wall clock" time)--for example:

  ```sql
  '2024-01-01 00:00:00'::TIMESTAMP AT TIME ZONE 'America/Los_Angeles'
  
  -- Returns
  2024-01-01T00:00:00-08:00
  ```

- When using an input timestamp with a timezone, both the `tz()` function and the `AT TIME ZONE`
  operator return the timestamp converted to the time in the specified timezone--for example:

  ```sql
  '2024-01-01T00:00:00-00:00' AT TIME ZONE 'America/Los_Angeles'
  tz('2024-01-01T00:00:00-00:00', 'America/Los_Angeles')
  
  -- Both return
  2023-12-31T16:00:00-08:00
  ```

- `tz()` always converts the input timestamp to the specified time zone.
  If the input timestamp does not have a timezone, the function assumes it is a UTC timestamp--for example:

  ```sql
  tz('2024-01-01 00:00:00'::TIMESTAMP, 'America/Los_Angeles')

  -- Returns
  2023-12-31T16:00:00-08:00
  ```

  ```sql
  tz('2024-01-01T00:00:00+1:00', 'America/Los_Angeles')

  -- Returns
  2023-12-31T15:00:00-08:00
  ```

{{< expand-wrapper >}}
{{% expand "View `tz` and `::timestamp` comparison" %}}

```sql
SELECT
  '2024-04-01T00:00:20Z'::timestamp AT TIME ZONE 'Europe/Brussels' as time_timestamp,
  tz('2024-04-01T00:00:20', 'Europe/Brussels') as time_tz;
```

| time_timestamp               | time_tz                    |
| :--------------------------- | :------------------------- |
| 2024-04-01T00:00:20+02:00    | 2024-04-01T02:00:20+02:00  |

{{% /expand %}}
{{< /expand-wrapper >}}
~~~

File: content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md [hunk 2]
```diff
@@ -1072,3 +1073,70 @@ SELECT
...
 {{% /expand %}}
 {{< /expand-wrapper >}}
+
+
+## tz
+
+Converts a timestamp to a provided timezone. If the second argument is not provided, it defaults to UTC.
+
+```sql
+tz(time[, timezone])
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md:1083]:
```suggestion
tz(time_expression[, timezone])
```

File: content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md [hunk 3]
```diff
@@ -1072,3 +1073,70 @@ SELECT
...
+```
+
+##### Arguments
+
+- **expression**: time to operate on.
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md:1088]:
```suggestion
- **time_expression**: Time to operate on.
```

File: content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md [hunk 4]
```diff
@@ -1072,3 +1073,70 @@ SELECT
...
+  '2024-04-01T00:00:20Z'::timestamp AT TIME ZONE 'Europe/Brussels' as time_timestamp,
+  tz('2024-04-01T00:00:20', 'Europe/Brussels') as time_tz;
+```
+
+| time_timestamp               | time_tz                    |
+| :--------------------------- | :------------------------- |
+| 2024-04-01T00:00:20+02:00    | 2024-04-01T02:00:20+02:00  |
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md:1142]:
~~~suggestion
##### Differences between tz and AT TIME ZONE

`tz` and [`AT TIME ZONE`](/influxdb/cloud-serverless/reference/sql/operators/other/#at-time-zone)
differ when the input timestamp **does not** have a timezone.

- When using an input timestamp that does not have a timezone (the default behavior in InfluxDB) with the
  `AT TIME ZONE` operator, the operator returns the the same timestamp, but with a timezone offset
  (also known as the "wall clock" time)--for example:

  ```sql
  '2024-01-01 00:00:00'::TIMESTAMP AT TIME ZONE 'America/Los_Angeles'
  
  -- Returns
  2024-01-01T00:00:00-08:00
  ```

- When using an input timestamp with a timezone, both the `tz()` function and the `AT TIME ZONE`
  operator return the timestamp converted to the time in the specified timezone--for example:

  ```sql
  '2024-01-01T00:00:00-00:00' AT TIME ZONE 'America/Los_Angeles'
  tz('2024-01-01T00:00:00-00:00', 'America/Los_Angeles')
  
  -- Both return
  2023-12-31T16:00:00-08:00
  ```

- `tz()` always converts the input timestamp to the specified time zone.
  If the input timestamp does not have a timezone, the function assumes it is a UTC timestamp--for example:

  ```sql
  tz('2024-01-01 00:00:00'::TIMESTAMP, 'America/Los_Angeles')

  -- Returns
  2023-12-31T16:00:00-08:00
  ```

  ```sql
  tz('2024-01-01T00:00:00+1:00', 'America/Los_Angeles')

  -- Returns
  2023-12-31T15:00:00-08:00
  ```

{{< expand-wrapper >}}
{{% expand "View `tz` and `::timestamp` comparison" %}}

```sql
SELECT
  '2024-04-01T00:00:20Z'::timestamp AT TIME ZONE 'Europe/Brussels' as time_timestamp,
  tz('2024-04-01T00:00:20', 'Europe/Brussels') as time_tz;
```

| time_timestamp               | time_tz                    |
| :--------------------------- | :------------------------- |
| 2024-04-01T00:00:20+02:00    | 2024-04-01T02:00:20+02:00  |

{{% /expand %}}
{{< /expand-wrapper >}}
~~~

File: content/influxdb/clustered/reference/sql/functions/time-and-date.md [hunk 5]
```diff
@@ -1072,3 +1073,70 @@ SELECT
...
 {{% /expand %}}
 {{< /expand-wrapper >}}
+
+
+## tz
+
+Converts a timestamp to a provided timezone. If the second argument is not provided, it defaults to UTC.
+
+```sql
+tz(time[, timezone])
```
Comment (sanderson) [inline content/influxdb/clustered/reference/sql/functions/time-and-date.md:1083]:
```suggestion
tz(time_expression[, timezone])
```

File: content/influxdb/clustered/reference/sql/functions/time-and-date.md [hunk 6]
```diff
@@ -1072,3 +1073,70 @@ SELECT
...
+```
+
+##### Arguments
+
+- **expression**: time to operate on.
```
Comment (sanderson) [inline content/influxdb/clustered/reference/sql/functions/time-and-date.md:1088]:
```suggestion
- **time_expression**: Time to operate on.
```

File: content/influxdb/clustered/reference/sql/functions/time-and-date.md [hunk 7]
```diff
@@ -1072,3 +1073,70 @@ SELECT
...
+  Can be a constant, column, or function, and any combination of arithmetic operators.
+- **timezone**: Optional [timezone db string](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) into which the value should be cast. Defaults to "UTC" if not provided
+  The function returns the timestamp cast to the correct timezone.
+  If an incorrect timezone string is passed, or the wrong datatype is provided, the function returns an error.
```
Comment (sanderson) [inline content/influxdb/clustered/reference/sql/functions/time-and-date.md:1092]:
```suggestion
- **timezone**: [Timezone string](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)
  to cast the value into. Default is `'UTC'`.
  The function returns the timestamp cast to the specified timezone.
  If an incorrect timezone string is passed or the wrong datatype is provided, the function returns an error.
```

File: content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md [hunk 8]
```diff
@@ -1072,3 +1073,70 @@ SELECT
...
+tz(time[, timezone])
+```
+
+##### Arguments
+
+- **expression**: time to operate on.
+  Can be a constant, column, or function, and any combination of arithmetic operators.
+- **timezone**: Optional [timezone db string](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) into which the value should be cast. Defaults to "UTC" if not provided
+  The function returns the timestamp cast to the correct timezone.
+  If an incorrect timezone string is passed, or the wrong datatype is provided, the function returns an error.
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/reference/sql/functions/time-and-date.md:1092]:
```suggestion
- **timezone**: [Timezone string](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)
  to cast the value into. Default is `'UTC'`.
  The function returns the timestamp cast to the specified timezone.
  If an incorrect timezone string is passed or the wrong datatype is provided, the function returns an error.
```

File: content/influxdb/clustered/reference/sql/functions/time-and-date.md [hunk 9]
```diff
@@ -1072,3 +1073,70 @@ SELECT
...
+  '2024-04-01T00:00:20Z'::timestamp AT TIME ZONE 'Europe/Brussels' as time_timestamp,
+  tz('2024-04-01T00:00:20', 'Europe/Brussels') as time_tz;
+```
+
+| time_timestamp               | time_tz                    |
+| :--------------------------- | :------------------------- |
+| 2024-04-01T00:00:20+02:00    | 2024-04-01T02:00:20+02:00  |
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
```
Comment (sanderson) [inline content/influxdb/clustered/reference/sql/functions/time-and-date.md:1142]:
~~~suggestion
##### Differences between tz and AT TIME ZONE

`tz` and [`AT TIME ZONE`](/influxdb/clustered/reference/sql/operators/other/#at-time-zone)
differ when the input timestamp **does not** have a timezone.

- When using an input timestamp that does not have a timezone (the default behavior in InfluxDB) with the
  `AT TIME ZONE` operator, the operator returns the the same timestamp, but with a timezone offset
  (also known as the "wall clock" time)--for example:

  ```sql
  '2024-01-01 00:00:00'::TIMESTAMP AT TIME ZONE 'America/Los_Angeles'
  
  -- Returns
  2024-01-01T00:00:00-08:00
  ```

- When using an input timestamp with a timezone, both the `tz()` function and the `AT TIME ZONE`
  operator return the timestamp converted to the time in the specified timezone--for example:

  ```sql
  '2024-01-01T00:00:00-00:00' AT TIME ZONE 'America/Los_Angeles'
  tz('2024-01-01T00:00:00-00:00', 'America/Los_Angeles')
  
  -- Both return
  2023-12-31T16:00:00-08:00
  ```

- `tz()` always converts the input timestamp to the specified time zone.
  If the input timestamp does not have a timezone, the function assumes it is a UTC timestamp--for example:

  ```sql
  tz('2024-01-01 00:00:00'::TIMESTAMP, 'America/Los_Angeles')

  -- Returns
  2023-12-31T16:00:00-08:00
  ```

  ```sql
  tz('2024-01-01T00:00:00+1:00', 'America/Los_Angeles')

  -- Returns
  2023-12-31T15:00:00-08:00
  ```

{{< expand-wrapper >}}
{{% expand "View `tz` and `::timestamp` comparison" %}}

```sql
SELECT
  '2024-04-01T00:00:20Z'::timestamp AT TIME ZONE 'Europe/Brussels' as time_timestamp,
  tz('2024-04-01T00:00:20', 'Europe/Brussels') as time_tz;
```

| time_timestamp               | time_tz                    |
| :--------------------------- | :------------------------- |
| 2024-04-01T00:00:20+02:00    | 2024-04-01T02:00:20+02:00  |

{{% /expand %}}
{{< /expand-wrapper >}}
~~~

**Review by sanderson — CHANGES_REQUESTED** [review]
@jeffreyssmith2nd Thanks for this! I added some suggestions.

File: content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md (hunk 1 above)
Comment (jeffreyssmith2nd) [reply content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md:1142]: Ah I like this way of describing it, much clearer!

**Pull request merged**